src/docs/development/platform-integration/add-to-app-android/add-flutter-fragment.md

```diff
@@ -0,0 +1,311 @@
+        flutterFragment.onUserLeaveHint();
+    }
+
+    @Override
+    public void onTrimMemory(int level) {
+        super.onTrimMemory(level);
+        flutterFragment.onTrimMemory(level);
+    }
+}
+```
+
+With the above OS signals forwarded to Flutter, your `FlutterFragment` works as
+expected. You have now added a `FlutterFragment` to your existing Android app.
+
+## Using a pre-warmed `FlutterEngine`
+
+By default, a `FlutterFragment` creates its own instance of a `FlutterEngine`,
+which requires non-trivial warmup time. This means your user sees a blank
+`Fragment` for a second or two. You can mitigate most of this warmup time by
+using an existing, pre-warmed instance of `FlutterEngine`.
+
+Please see the [instructions for instantiating and starting a `FlutterEngine`].
+
+[instructions for instantiating and starting a `FlutterEngine`]: /docs/development/platform-integrations/add-to-app-android/flutter-engine
+
+To use a pre-warmed `FlutterEngine` in a `FlutterFragment`, subclass
+`FlutterFragment` and override `provideFlutterEngine()` as shown below:
+
+```java
+public class MyFlutterFragment extends FlutterFragment {
+    @Override
+    protected FlutterEngine provideFlutterEngine(@NonNull Context context) {
+        // Obtain a reference to, and return, your pre-warmed FlutterEngine.
+        // For example:
+        return ((MyApplication) context.getApplication())
+            .getPreWarmedFlutterEngine();
+    }
+
+    // When providing a pre-warmed engine within a FlutterFragment subclass,
+    // be sure to override this method and return "true" so that this
+    // FlutterFragment does not destroy your FlutterEngine when the Fragment
+    // is destroyed.
+    @Override
+    protected boolean retainFlutterEngineAfterFragmentDestruction() {
+        return true;
+    }
+}
+```
+
+By providing a pre-warmed `FlutterEngine` as shown above, your app renders the
+first Flutter frame as quickly as possible.
+
+## Display a splash screen
+
+The initial display of Flutter content requires some amount of time, even if a
+pre-warmed `FlutterEngine` is used. To help improve the user experience around
+this brief waiting period, Flutter supports the display of a splash screen until
+Flutter renders its first frame. For instructions about showing a splash screen,
+please see the [Android splash screen guide].
+
+[Android splash screen guide]: /docs/development/platform-integration/add-to-app-android/splash-screens
+
+## Run Flutter with a specified initial route
+
+An Android app might contain many independent Flutter experiences, running in
+different `FlutterFragment`s, with different `FlutterEngine`s. In these
+scenarios, it is common for each Flutter experience to begin with different
+initial routes (routes other than `/`). To facilitate this, `FlutterFragment`'s
+`Builder` allows you to specify a desired initial route, as shown below:
+
+```java
+FlutterFragment flutterFragment = new FlutterFragment.Builder()
```

Comment:
I think this might not work if you're using a prewarmed engine since you already called run dart entrypoint

Reply:
That's correct. Initial route will have no effect on a pre-warmed engine. Same for `FlutterActivity`.

Reply:
Can we add an info bubble for this caveat?

Reply:
Sure. Added a bubble.